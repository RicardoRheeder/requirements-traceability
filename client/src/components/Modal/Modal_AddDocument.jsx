import React, { useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { useAuth0 } from '@auth0/auth0-react'
import { createDocAsync } from '../../redux/stores/documents/actions'

export default function Modal_AddDocument() {
<<<<<<< HEAD
  const dispatch = useDispatch()
  const { user } = useAuth0()

  const documents = useSelector((state) => state.documents.documents)
  const error = useSelector((state) => state.documents.error)

  const [inputData, setInputData] = useState('')

  console.log(documents)

  const submitName = () => {
    // slicing the user id to remove "auth0|"
    const oldId = user.sub
    const newID = oldId.slice(6, oldId.length)

    // createDocAsync({ title: 'test123', admin: newID })

    dispatch(createDocAsync({ title: inputData, admin: newID }))
  }

  return (
    <div>
      <h2>Add document</h2>
      <div>Please enter the name of the empty document to create.</div>
      <form>
        <input
          value={inputData}
          onChange={(e) => setInputData(e.target.value)}
        />
        <button onClick={submitName}>Submit</button>
      </form>
=======
  return (
    <div className="modal-root modal-root-child">
      <div className="modal-contents-container">
        <h1 className="modal-contents-title">Add document</h1>
        <h2>Please enter the name of the empty document to create.</h2>
        <form>
          <input className="modal-input" />
          <button className="orange-button">Submit</button>
        </form>
      </div>
>>>>>>> master
    </div>
  )
}
