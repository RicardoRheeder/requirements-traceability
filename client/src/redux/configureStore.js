import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

// restricting extension in production using process.env.NODE_ENV which is set 'production'
import { composeWithDevTools } from 'redux-devtools-extension/logOnlyInProduction';

import store from './stores';

const enhancer = (process.env.NODE_ENV == 'development')
               ? composeWithDevTools(applyMiddleware(thunk))
               : applyMiddleware(thunk);

export default () => createStore(store, enhancer);
