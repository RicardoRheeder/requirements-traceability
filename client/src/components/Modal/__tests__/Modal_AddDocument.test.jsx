import React from "react"
import Enzyme, { shallow, mount} from "enzyme"
import Modal_AddDocument from "../Modal_AddDocument"
import Adapter from "enzyme-adapter-react-16"
import { Provider } from "react-redux";
import configureStore from "redux-mock-store"


const mockStore = configureStore([])
Enzyme.configure({adapter: new Adapter() });

describe("Add Document modal component", () => {
    let store
    beforeEach(() => {
        store = mockStore({user:{
            info: null
        }});
    });
    test("renders", () => {
        const wrapper = shallow(<Provider store = {store}><Modal_AddDocument/></Provider>);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
    test("renders with children", () => {
        const wrapper = mount(<Provider store = {store}><Modal_AddDocument/></Provider>);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
});