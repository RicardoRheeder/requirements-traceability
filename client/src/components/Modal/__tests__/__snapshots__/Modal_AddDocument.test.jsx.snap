// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Document modal component renders 1`] = `ShallowWrapper {}`;

exports[`Add Document modal component renders with children 1`] = `ReactWrapper {}`;
